apis/python/src/tiledbsoma/io/shaping.py

```diff
@@ -378,9 +468,10 @@ def _recurse(
             retval["var"] = _recurse(item, node_name="var", **kwargs)
             for iter_coll_name in ["X", "obsm", "obsp", "varm", "varp"]:
                 kwargs["coll_name"] = iter_coll_name
-                retval[iter_coll_name] = _recurse(
-                    item, node_name=iter_coll_name, **kwargs
-                )
+                children_dict = _recurse(item, node_name=iter_coll_name, **kwargs)
+                # If the experiment has no varp (etc.) at all, don't include an entry for it.
```

Comment:
Should we add tests for running on an empty `Experiment` and empty `Measurement`? (Could be a follow-up story for later).

Reply:
Hm, I think that's legit to include in this PR -- 👀

Reply:
Implemented on #3966